token/sdk/tms/db/tms.go

```diff
@@ -137,9 +146,18 @@ func (p *PostInitializer) ConnectNetwork(networkID, channel, namespace string) e
 	if err := n.ProcessorManager().AddProcessor(
 		namespace,
 		fabric2.NewTokenRWSetProcessor(
-			n,
+			n.Name(),
 			namespace,
-			p.sp,
+			func() *token3.ManagementServiceProvider {
+				return token3.GetManagementServiceProvider(p.sp)
+			},
+			func(tms *token3.ManagementService, txID string) ([]byte, error) {
```

Comment:
I may be missing something, but I think we could create a function and reuse it for both constructors.